src/ActivityLoggerBatch.php

```diff
@@ -0,0 +1,54 @@
+<?php
+namespace Spatie\Activitylog;
+
+use Closure;
+use Ramsey\Uuid\Uuid;
+
+class ActivityLoggerBatch
+{
+    public ?string $uuid = null;
+
+    public int $transactions = 0;
+
+    protected function generateUUID(): string
+    {
+        return Uuid::uuid4()->toString();
+    }
+
+    public function getUuid(): ?string
+    {
+        return $this->uuid;
+    }
+
+    public function useBatch(Closure $callback): mixed
```

Comment:
Would be `withinBatch()` a better naming?
The model has a `withoutEvents()` and we have a `withoutLogs()` - so I think that `within` matches better the naming pattern than `use` does.